python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1396,6 +1463,845 @@ def test_nccl_all_reduce_with_class_method_output_node(ray_start_regular):
         assert result == [t1 + t4, t1 + t4, t2, t3]
 
 
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_reduce_scatter(ray_start_regular):
```

Comment:
For allgather and reducescatter, only keep two tests:
1. the most basic one "test_torch_tensor_nccl_all_reduce".
2. the most complex one "test_torch_tensor_nccl_all_reduce_scheduling".
Then remove the others.

Also rename "test_nccl_all_reduce_with_class_method_output_node" to "test_torch_tensor_nccl_all_reduce_with_class_method_output_node".